bot/src/main/java/com/community/tools/service/TrackingService.java

```diff
@@ -15,64 +15,48 @@
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 import lombok.RequiredArgsConstructor;
-import org.springframework.beans.factory.annotation.Autowired;
+import net.dv8tion.jda.api.entities.Message;
 import org.springframework.statemachine.StateMachine;
 import org.springframework.stereotype.Service;
 
 @Service
 @RequiredArgsConstructor
 public class TrackingService {
 
-  @Autowired
-  private MessageService messageService;
-
-  @Autowired
-  private StateMachineService stateMachineService;
-
-  @Autowired
-  private UserRepository userRepository;
-
-  @Autowired
-  private EstimateTaskService estimateTaskService;
-
-  @Autowired
-  private MessageConstructor messageConstructor;
-
-  @Autowired
-  private EmailService emailService;
+  private final MessageService<?> messageService;
+  private final StateMachineService stateMachineService;
+  private final UserRepository userRepository;
+  private final EstimateTaskService estimateTaskService;
+  private final EmailService emailService;
 
   /**
    * Method to start the event by state.
    *
-   * @param messageFromUser message from user
-   * @param userId          user id
+   * @param message message from user
    * @throws Exception Exception
    */
-  public void doAction(String messageFromUser, String userId) throws Exception {
+  public void doAction(Message message) throws Exception {
+    String userId = message.getAuthor().getId();
+    String messageFromUser = message.getContentRaw();
 
     StateMachine<State, Event> machine = stateMachineService.restoreMachine(userId);
     String userForQuestion = machine.getExtendedState().getVariables().get("id").toString();
 
-    String message = Messages.DEFAULT_MESSAGE;
+    String messageToSend = Messages.DEFAULT_MESSAGE;
     Event event = null;
     Payload payload = null;
 
     switch (machine.getState().getId()) {
       case NEW_USER:
```

Comment:
It's better to decouple your logic from this StateMachine. We'd like to get rid of this service in the future.